### ISSUE TITLE
Decrease file size by storing colors in a table

### ISSUE THREAD

**Issue by schitcrafter** [mbrlabs/Lorien#18]
I realize that this is not something the project needs currently, but I had the idea for it so I'm suggesting it.

# Basic concept:

Instead of storing color separately for every stroke:

- store all colors in a table
- hold an index to the table in every stroke

# Further improvements

Allocate a region of the indices (say, 0x00 - 0xFF) for predetermined, heavily used colors (Ideally, the ones being suggested by the color picker).

# Implementation

- null-terminated table
- store a value containing the size of the indices (i.e. 8 for uint8, 16 for uint16) so we don't store unnecessary 0's while also allowing for infinitely many colors

**Comment by mbrlabs** [comment]
Yes good idea, but i'd like to delay optimizations like that until i'm confident enough that the file format won't change much.
